File: app/src/main/java/com/example/dr/turis_huma/RegisterActivity.java
package com.example.dr.turis_huma;

import android.app.ProgressDialog;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;
import android.text.TextUtils;
import android.view.View;
import android.widget.*;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;

public class RegisterActivity extends AppCompatActivity {
    private EditText userEmail,userPassword,userConfirmPassword,userUsername,userCountry;
    private ImageButton createAccountButton;
    private ProgressDialog loadingBar;
    private FirebaseAuth mAuth;

    @Override
    protected void onCreate(Bundle savedInstanceState){
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_register);

        userEmail = findViewById(R.id.txtEmail);
        userUsername = findViewById(R.id.txtNombre);
        userPassword = findViewById(R.id.txtPassword);
        userConfirmPassword = findViewById(R.id.txtPasswordConfirm);
        userCountry = findViewById(R.id.txtCountry);
        createAccountButton = findViewById(R.id.register_next_button);
        loadingBar = new ProgressDialog(this);

        mAuth= FirebaseAuth.getInstance();

        createAccountButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                CreateNewAccount();
            }
        });

    }

    private void CreateNewAccount() {
        String email = userEmail.getText().toString().trim();
        String username = userUsername.getText().toString().trim();
        String password = userPassword.getText().toString().trim();
        String passwordConfirm =userConfirmPassword.getText().toString().trim();
        String country = userCountry.getText().toString().trim().toUpperCase();

        if(TextUtils.isEmpty(username)){
            Toast.makeText(this,"Debe ingresar un nombre usuario",Toast.LENGTH_SHORT).show();
        }else if(TextUtils.isEmpty(email)){
            Toast.makeText(this,"Debe ingresar un correo electrónico",Toast.LENGTH_SHORT).show();
        }else if(TextUtils.isEmpty(password)){
            Toast.makeText(this,"Debe ingresar una contraseña",Toast.LENGTH_SHORT).show();
        }else if(TextUtils.isEmpty(passwordConfirm)){
            Toast.makeText(this,"Debe confirmar la contraseña",Toast.LENGTH_SHORT).show();
        }else if(!password.equals(passwordConfirm)){
            Toast.makeText(this,"Las contraseñas no coinciden",Toast.LENGTH_SHORT).show();
        }else if (TextUtils.isEmpty(country)){
            Toast.makeText(this,"Debe ingresar un país",Toast.LENGTH_SHORT).show();
        }else {

            loadingBar.setTitle("Creando cuenta");
            loadingBar.setMessage("Espere mientras creamos su cuenta");
            loadingBar.show();
            loadingBar.setCanceledOnTouchOutside(true);

            mAuth.createUserWithEmailAndPassword(email,password).addOnCompleteListener(new OnCompleteListener<AuthResult>() {
                @Override
                public void onComplete(@NonNull Task<AuthResult> task) {
                    if (task.isSuccessful()) {
                        SendUserToLogin();
                        Toast.makeText(RegisterActivity.this, "Se ha autenticado correctamente", Toast.LENGTH_SHORT).show();
                        loadingBar.dismiss();
                    }else{
                        String message =  task.getException().getMessage();
                        Toast.makeText(RegisterActivity.this,"Error: "+message,Toast.LENGTH_SHORT).show();
                        loadingBar.dismiss();
                    }
                }
            });



        }
    }

    private void SendUserToLogin() {
        Intent loginIntent = new Intent(RegisterActivity.this,LoginActivity.class);
        loginIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
        startActivity(loginIntent);
        finish();
    }
}

File: app/src/main/java/com/example/dr/turis_huma/RegisterActivity.kt
package com.example.dr.turis_huma

import android.app.ProgressDialog
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.text.TextUtils
import android.widget.Button
import android.widget.EditText
import android.widget.ImageButton
import android.widget.Toast
import com.google.android.gms.tasks.Task
import kotlinx.android.synthetic.*
import kotlinx.android.synthetic.main.activity_login.*
import kotlinx.android.synthetic.main.activity_register.view.*
import com.google.firebase.auth.FirebaseAuth
import kotlinx.android.synthetic.main.activity_register.*


class RegisterActivity : AppCompatActivity() {

    // private val mAuth: FirebaseAuth = FirebaseAuth.getInstance()
    // private val loadingBar:ProgressDialog = ProgressDialog.show(this,"Creando nueva cuenta","Espere mientras creamos su cuenta")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_register)

        val userName:EditText = txtNombre
        val userEmail:EditText = findViewById(R.id.txtEmail)
        val userPassword :EditText= findViewById(R.id.txtPassword)
        val userConfirmPassword :EditText= findViewById(R.id.txtPasswordConfirm)
        val userCountry :EditText= findViewById(R.id.txtCountry)
        //val createAccountButton : ImageButton = findViewById(R.id.btnNext2)



       // createAccountButton.setOnClickListener(){
           // createNewAccount(userName, userEmail, userPassword, userConfirmPassword,userCountry)
        //}
    }


   /* fun createNewAccount(username:EditText, email:EditText, pass:EditText, confPass:EditText, country:EditText){
        val userValue = username.text.toString()
        val userEmail = email.text.toString()
        val userPassword = pass.text.toString()
        val userConfPass = confPass.text.toString()
        val userCountry = country.text.toString()

        if(TextUtils.isEmpty(userValue)){
            Toast.makeText(this,"Debe ingresar un nombre usuario",Toast.LENGTH_SHORT).show()
        }else if(TextUtils.isEmpty(userEmail)){
            Toast.makeText(this,"Debe ingresar un correo electrónico",Toast.LENGTH_SHORT).show()
        }else if(TextUtils.isEmpty(userPassword)){
            Toast.makeText(this,"Debe ingresar una contraseña",Toast.LENGTH_SHORT).show()
        }else if(TextUtils.isEmpty(userConfPass)){
            Toast.makeText(this,"Debe confirmar la contraseña",Toast.LENGTH_SHORT).show()
        }else if(!userPassword.equals(userConfPass)){
            Toast.makeText(this,"Las contraseñas no coinciden",Toast.LENGTH_SHORT).show()
        }else if (TextUtils.isEmpty(userCountry)){
            Toast.makeText(this,"Debe ingresar un correo electrónico",Toast.LENGTH_SHORT).show()
        }else{

            loadingBar
            loadingBar.setCanceledOnTouchOutside(true)
            mAuth.createUserWithEmailAndPassword(userEmail,userPassword).addOnCompleteListener{
                task ->
                if (task.isSuccessful){
                    Toast.makeText(this,"El usuario ha sido creado exitosamente",Toast.LENGTH_SHORT).show()
                    loadingBar.dismiss()
                }else{
                    val message:String = task.exception?.message.toString()
                    Toast.makeText(this,"Error: ${message}",Toast.LENGTH_SHORT).show()
                    loadingBar.dismiss()
                }
            }
        }
    }
*/

}
